javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -760,7 +760,7 @@ public T head() {
         if (isEmpty()) {
             throw new NoSuchElementException("head of empty Vector");
         } else {
-            return get(0);
+            return getAt(leading, leadingOffset);
```

Comment:
the `leading` has elements always, making the `get(0)` slightly faster (~30%, compared to `get(0)`)